lib/Doctrine/Common/Cache/RedisCache.php

```diff
@@ -57,8 +57,11 @@ protected function doFetchMultiple(array $keys)
         $foundItems = [];
 
         foreach ($fetchedItems as $key => $value) {
-            if ($value === false && ! $this->redis->exists($key)) {
-                continue;
+            if ($value === false) {
+                if ( ! $this->redis->exists($key)) {
```

Comment:
There is still race condition between `->exists()` and `->get()` calls, this would require different approach (either MULTI or relying on GET only).

Reply:
Nvm, this just reduces the likeliness but does indeed not remove the race condition.